### PR TITLE
GetLabel: Handle correctly string value lookling like a transform pattern

### LINKED ISSUE #957: StringItems with a state containing "():" cause transform warnings
Example Item...
```
Switch	US_Hallway_Ceiling_Motion	                "Hallway Ceiling (upstairs) [MAP(motion.map):%s]"	                <motion>	    (gUS_Hallway,gStairwell_Trigger,gUS_Hallway_Trigger,gMotion_Sensor,gSleep_Security)		                                            {channel="zwave:device:55555:node5:sensor_binary"}
```
Example log entry...
```
2019-08-04 23:19:52.927 [WARN ] [org.eclipse.smarthome.ui.internal.items.ItemUIRegistryImpl] - couldn't transform value in label because transformationService of type 'Hallway Ceiling ' is unavailable
```
I thought this started after the ESH reintegration, but I could be wrong and there don't seem to be any changes to the file that would have caused this. It does not seem to effect functionality, but it's an annoying log that happens many times a day. I have several Items with parenthesis in the name, but these Items only appear in a sitemap as a Group. I haven't been able to track down when this occurs. I thought it may be linked to the use of a UI, but it does not happen every time.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #957

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by openhab-5iver** [comment]
I tested this and still got the warning in the log. :frowning_face:

**Comment by lolodomo** [comment]
@openhab-5iver : you can try the new PR if you like.

**Comment by openhab-5iver** [comment]
Tested and I no longer see the warning! Thank you again!

**Comment by cweitkamp** [comment]
I appreciate a solution for this :+1:. Afaik we get the warning `couldn't transform value in label because transformationService of type 'MAP' is unavailable` if one tries to use the MAP transformation without the bundle being installed. Can you assert that the warning will still appear in this case?

**Comment by openhab-5iver** [comment]
After removing the MAP transformation service, I did not see any warning, which I expected. After then saving an items file containing a test Item that used MAP, I got this warning for every one of my Items that uses MAP...
```
2019-08-08 00:15:04.364 [WARN ] [org.eclipse.smarthome.io.rest.core.item.EnrichedItemDTOMapper] - Failed transforming the state 'ON' on item 'Test_Map' with pattern 'MAP(motion.map):%s': Couldn't transform value because transformation service of type 'MAP' is not available.
```
Sending a command to the Item did not cause an error or warning.

In a DSL rule, this line...
```
logWarn("Rules", "Test [{}]", transform("MAP", "motion.map", "ON"))
```
... returned...
```
2019-08-08 00:27:46.446 [WARN ] [org.eclipse.smarthome.model.script.Rules] - Test [ON]
```
... without any warnings or errors.

**Comment by lolodomo** [comment]
I have of course not removed the warning that will appear if the transform service is not available.

**Comment by lolodomo** [comment]
I see the warning when I run the unit tests.

File: bundles/org.openhab.core.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java [hunk 1]
```diff
@@ -498,11 +510,12 @@ protected String formatUndefined(String formatPattern) {
      * If the value does not start with the call to a transformation service,
      * we return the label with the mapped option value if provided (not null).
      */
-    private String transform(String label, String labelMappedOption) {
+    private String transform(String label, boolean matchTransform, String labelMappedOption) {
         String ret = label;
-        if (getFormatPattern(label) != null) {
-            Matcher matcher = EXTRACT_TRANSFORMFUNCTION_PATTERN.matcher(label);
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.core.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: You are missing nothing :)

**Pull request merged**